interpreter/cling/lib/Interpreter/InvocationOptions.cpp

```diff
@@ -152,12 +156,22 @@ void CompilerOptions::Parse(int argc, const char* const argv[],
         CUDAHost = false;
         break;
 
+      case options::OPT_mllvm:
+        LLVMArgs.push_back(arg->getValue());
+        break;
+
       default:
         if (Inputs && arg->getOption().getKind() == Option::InputClass)
           Inputs->push_back(arg->getValue());
         break;
     }
   }
+
+  // Check that there were LLVM arguments, other than the first dummy entry.
+  if (LLVMArgs.size() > 1) {
+    LLVMArgs.push_back(nullptr);
+    llvm::cl::ParseCommandLineOptions(LLVMArgs.size() - 1, LLVMArgs.data());
```

Comment:
Why we can't pass the arguments through here?

Reply:
Not sure what you mean: We pass the arguments collected from `-mllvm`.

Reply:
I guess my question was how -mllvm is different than the rest of the arguments which are handled without ParseCommandLineOptions

Reply:
Clang options are parsed via their own infrastructure, I think this happens in the CIFactory. For `-mllvm`, arguments have to go through LLVM's infrastructure (`ParseCommandLineOptions`). FWIW Clang does the same https://github.com/root-project/root/blob/87a998d48803bc207288d90038e60ff148827664/interpreter/llvm/src/tools/clang/lib/FrontendTool/ExecuteCompilerInvocation.cpp#L228-L240 (which is where I based this code on).

Reply:
I see what's the problem now. I think we should fix the `llvm::cl` option parsing rather than avoiding to use it. Maybe the positional arguments should be considered depending on the `llvm::cl::cat` settings. In that case the 'global' rootcling arguments will live in their own 'namespace'.

Reply:
Yes, this can already be achieved with subcommands. The problem is that then `rootcling` would need to force a subcommand, which I haven't found a reliable way of doing. Instead, we can make the positional arguments only mandatory if we know that we are in `rootcling`. Then the options are always registered, but shouldn't cause much harm for parsing `-mllvm`.

Reply:
I am not sure if that would require introducing a new subcommand. If so, that would be a breaking change and the experiments will probably be not thrilled...

Reply:
If you can make it working, great. Otherwise the current solution works and doesn't change any user-visible stuff, apart from added support for `-mllvm` argument parsing.